### PR TITLE
CM-287:Add the payment method to indicate the way how payment should be processed

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://openimis.atlassian.net/browse/CM-287

### CODE REVIEW

File: payroll/gql_queries.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ class Meta:
         filter_fields = {
             "id": ["exact"],
             "name": ["iexact", "istartswith", "icontains"],
+            "status": ["exact", "iexact", "startswith", "istartswith", "contains", "icontains"],
+            "payment_method": ["exact", "iexact", "startswith", "istartswith", "contains", "icontains"],
```
Comment (malinowskikam) [inline payroll/gql_queries.py]: Do we need both case sensitive and insensitive filters?

File: payroll/apps.py [hunk 2]
```diff
@@ -46,4 +48,17 @@ def __load_config(cls, cfg):
...
+            StrategyMobilePayment,
+            StrategyOnSitePayment
+        )
+        PaymentsMethodRegistryPoint.register_payment_method(
+            payment_method_class_list=[
+                StrategyOnlinePayment(),
+                StrategyMobilePayment(),
+                StrategyOnSitePayment()
+            ]
+        )
```
Comment (malinowskikam) [inline payroll/apps.py:66]: I'm not sure why there is extra code in __load_config method as it's only purpose is to load the config keys. I think this code (and the custom filters one) Should be in ready() method or in a new method.

File: payroll/strategies/strategy_of_payments_interface.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class StrategyOfPaymentInterface:
```
Comment (malinowskikam) [inline payroll/strategies/strategy_of_payments_interface.py]: We should add ABC as base

**Review by malinowskikam — CHANGES_REQUESTED** [review]
Great work, I left some comments

File: payroll/gql_queries.py (hunk 1 above)
Comment (sniedzielski) [reply payroll/gql_queries.py]: I think if we have ENUM statuses and methods - we could stay with having only case sensitive ones

**Comment by sniedzielski** [comment]
@malinowskikam please re-check PR.

File: payroll/apps.py (hunk 2 above)
Comment (sniedzielski) [reply payroll/apps.py:66]: Moved to separate method which is triggered in 'ready' method after another steps

File: payroll/strategies/strategy_of_payments_interface.py (hunk 3 above)
Comment (sniedzielski) [reply payroll/strategies/strategy_of_payments_interface.py]: ok

**Review by malinowskikam — APPROVED** [review]

**Pull request merged**